react/headless/src/utils/directive.ts

```diff
@@ -1,61 +1,59 @@
 import type {Directive, DirectiveAndParam} from '@agnos-ui/core/types';
-import {attributesData, mergeDirectives} from '@agnos-ui/core/utils/directive';
+import {attributesData, bindDirective, classDirective, mergeDirectives} from '@agnos-ui/core/utils/directive';
+import {writable} from '@amadeus-it-group/tansu';
 import {BROWSER} from 'esm-env';
 import type {RefCallback} from 'react';
 import {useCallback, useMemo, useRef} from 'react';
 
 export * from '@agnos-ui/core/utils/directive';
 
-export function useDirective(directive: Directive<void>): {ref: RefCallback<HTMLElement>};
-export function useDirective<T>(directive: Directive<T>, args: T): {ref: RefCallback<HTMLElement>};
 /**
- * The useDirective function.
- *
- * Allows to attach a provided directive to the current react component.
- *
- * @param directive - the directive
- * @param args - the args to pass to the directive
- * @returns the ref callback
+ * Returns a class directive.
+ * @param className - class name to use
+ * @returns a class directive, to be used with {@link useDirectives}.
  */
-export function useDirective<T>(directive: Directive<T>, args?: T): {ref: RefCallback<HTMLElement>} {
-	const instance = useRef<ReturnType<typeof directive>>();
-	const propsRef = useRef<T>();
-	const ref = useCallback(
-		(element: HTMLElement | null) => {
-			instance.current?.destroy?.();
-			instance.current = undefined;
-			if (element) {
-				instance.current = directive(element, propsRef.current as T);
-			}
-		},
-		[directive],
-	);
-	propsRef.current = args;
-	instance.current?.update?.(args as T);
-	return {ref};
-}
-
-export function useDirectives(directives: Directive<void>[]): {ref: RefCallback<HTMLElement>};
-export function useDirectives<T>(directives: Directive<T>[], args: T): {ref: RefCallback<HTMLElement>};
-/**
- * The useDirectives function.
- *
- * Allows to attach multiple directives to the current react component.
- *
- * @param directives - directives
- * @param args - the args to pass to the directives
- * @returns the ref callback
- */
-export function useDirectives<T>(directives: Directive<T>[], args?: T): {ref: RefCallback<HTMLElement>} {
-	const mergedDirectives = useMemo(() => mergeDirectives(...directives), directives);
-	return useDirective(mergedDirectives, args as any);
-}
+export const useClassDirective = (className: string) => {
+	const {directive, className$} = useMemo(() => {
+		const className$ = writable('');
+		return {directive: bindDirective(classDirective, className$), className$};
+	}, []);
+	className$.set(className);
+	return directive;
+};
 
 const attributesMap = new Map([
 	['tabindex', 'tabIndex'],
 	['for', 'htmlFor'],
 ]);
 
+// For boolean attributes, the presence of the attribute means true, but react wants a boolean value
+// cf the list in https://github.com/facebook/react/blob/bf40b024421a0e1f2f882fd7171ea39cd74c88df/packages/react-dom-bindings/src/client/ReactDOMComponent.js#L665
+const booleanAttributes = new Set([
```

Comment:
As discussed internally, this is not maintainable in the long-term.
Am ok to merge as-is, but please create a issue to track this, and hopefully later on we will have time for a future-proof solution (maybe even nothing to do, if React were to export this enum ?)

Reply:
I have opened #714 to track this